### PR TITLE
modified intensity_black_body using numba

### PR DESCRIPTION
**Pull request by shreyansjain012**
Optimize the `intensive_black_body` speed with numba
## Description

- Imported njit from numba
- Removed numexpr because numba can only work with numpy arrays
- Modified function intensive_black_body with numba
## Motivation and Context

It significantly reduces the processing speed of function

## How Has This Been Tested?

For testing of performance, I ran the commands

`from tardis import run_tardis`
`from tardis.io.atom_data.util import download_atom_data`
`download_atom_data('kurucz_cd23_chianti_H_He')`
`sim = run_tardis('tardis_example.yml')`

and then I profiled this function `run_tardis` so that I can see what other function it is calling and their respective time, from there I searched for the function `intensity_black_body` which gives the current output. I did this thing with and without modifying the function with numba
image

## Screenshots (if appropriate):
When I tried using `intensity_black_body` with my own input parameters in a separate notebook and profiled it I found that it does indeed optimizes the function but only after the second call, for the first call it takes more time than usual(without numba) so I am guessing it is because of low number of total number of times this function is called(21 times)
Screenshot from 2020-02-25 20-47-53
**without using numba**

Screenshot from 2020-02-25 20-48-09
**when using function with numba**

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: tardis/util/base.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 from tardis import constants
 from astropy import units as u
 from pyne import nucname
+from numba import njit
+from tqdm.autonotebook import tqdm
```
Comment (wkerzendorf) [inline tardis/util/base.py]: why do you import this?

**Comment by wkerzendorf** [comment]
okay - try to profile the whole code and see how much of a difference it makes

File: tardis/util/base.py (hunk 1 above)
Comment (shreyansjain012) [reply tardis/util/base.py]: actually I ran this on jupyter notebook and without this it was giving some warnings.

Comment (wkerzendorf) [reply tardis/util/base.py]: but you do not use tqdm in this function, right?

Comment (shreyansjain012) [reply tardis/util/base.py]: Yes I didn't. I will remove it.

**Comment by shreyansjain012** [comment]
> okay - try to profile the whole code and see how much of a difference it makes

Okay, thanks. Let me learn about code profiling and I' ll get back to you

**Comment by shreyansjain012** [comment]
@wkerzendorf `intensity_black_body` function is not called inside base.py where am I supposed to use cprofile?

**Comment by wkerzendorf** [comment]
follow the quickstart example - have a look what the functions are that call this.

**Comment by shreyansjain012** [comment]
@wkerzendorf 
When running j_blues.py for profiling `intensity_black_body` using numba on the contrary decreased the performance a little bit

performance when using numba
with numba

performance when not using numba
withoutNumba

**Comment by wkerzendorf** [comment]
you are just importing the module - not really running code in there. Have a look at the quickstart notebook - run the whole TARDIS simulation and then look from there where else to change code

**Comment by shreyansjain012** [comment]
@wkerzendorf I followed the quickstart notebook. Ran whole tardis simulation and got this stats

image

am I still doing something wrong?

**Comment by wkerzendorf** [comment]
what function did you profile?

**Comment by shreyansjain012** [comment]
> what function did you profile?

run_tardis

**Pull request closed without merging**